aws/resource_aws_sagemaker_app_test.go

```diff
@@ -0,0 +1,375 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/sagemaker/finder"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_sagemaker_app", &resource.Sweeper{
+		Name: "aws_sagemaker_app",
+		F:    testSweepSagemakerApps,
+	})
+}
+
+func testSweepSagemakerApps(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %w", err)
+	}
+	conn := client.(*AWSClient).sagemakerconn
+	var sweeperErrs *multierror.Error
+
+	err = conn.ListAppsPages(&sagemaker.ListAppsInput{}, func(page *sagemaker.ListAppsOutput, lastPage bool) bool {
+		for _, app := range page.Apps {
+
+			if aws.StringValue(app.Status) == sagemaker.AppStatusDeleted {
+				continue
+			}
+
+			r := resourceAwsSagemakerApp()
+			d := r.Data(nil)
+			d.SetId(aws.StringValue(app.AppName))
+			d.Set("app_name", app.AppName)
+			d.Set("app_type", app.AppType)
+			d.Set("domain_id", app.DomainId)
+			d.Set("user_profile_name", app.UserProfileName)
+			err := r.Delete(d, client)
+
+			if err != nil {
+				log.Printf("[ERROR] %s", err)
+				sweeperErrs = multierror.Append(sweeperErrs, err)
+				continue
+			}
+		}
+
+		return !lastPage
+	})
+
+	if testSweepSkipSweepError(err) {
+		log.Printf("[WARN] Skipping SageMaker domain sweep for %s: %s", region, err)
+		return sweeperErrs.ErrorOrNil()
+	}
+
+	if err != nil {
+		sweeperErrs = multierror.Append(sweeperErrs, fmt.Errorf("error retrieving Sagemaker Apps: %w", err))
+	}
+
+	return sweeperErrs.ErrorOrNil()
+}
+
+func testAccAWSSagemakerApp_basic(t *testing.T) {
+	var app sagemaker.DescribeAppOutput
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_sagemaker_app.test"
+
+	resource.ParallelTest(t, resource.TestCase{
```

Comment:
To get test serialization you will also have to change all the relevant `resource.ParallelTest(...)` calls into `resource.Test(...)` calls, otherwise I get errors like:

```console
$ SAGEMAKER_IMAGE_VERSION_BASE_IMAGE=x make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSSagemakerDomain_\|TestAccAWSSagemakerApp_\|TestAccAWSSagemakerUserProfile_disappears'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerDomain_\|TestAccAWSSagemakerApp_\|TestAccAWSSagemakerUserProfile_disappears -timeout 120m
=== RUN   TestAccAWSSagemakerDomain_serial
=== RUN   TestAccAWSSagemakerDomain_serial/Domain
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/basic
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/basic
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/tags
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/tags
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/tensorboardAppSettings
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/tensorboardAppSettings
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/kernelGatewayAppSettings_customImage
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/kernelGatewayAppSettings_customImage
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/jupyterServerAppSettings
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/jupyterServerAppSettings
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/kms
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/kms
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/disappears
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/disappears
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/tensorboardAppSettingsWithImage
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/tensorboardAppSettingsWithImage
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/kernelGatewayAppSettings
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/kernelGatewayAppSettings
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/securityGroup
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/securityGroup
=== RUN   TestAccAWSSagemakerDomain_serial/Domain/sharingSettings
=== PAUSE TestAccAWSSagemakerDomain_serial/Domain/sharingSettings
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/basic
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/sharingSettings
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/securityGroup
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/kernelGatewayAppSettings
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/tensorboardAppSettingsWithImage
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/disappears
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/kms
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/jupyterServerAppSettings
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/kernelGatewayAppSettings_customImage
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/tensorboardAppSettings
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/tags
=== CONT  TestAccAWSSagemakerDomain_serial/Domain/securityGroup
    resource_aws_sagemaker_domain_test.go:224: Step 1/3 error: Error running apply: 2021/02/04 07:51:43 [DEBUG] Using modified User-Agent: Terraform/0.12.26 HashiCorp-terraform-exec/0.12.0
        
        Error: error creating SageMaker domain: ResourceLimitExceeded: The account-level service limit 'Total domains' is 1 Domains, with current utilization of 1 Domains and a request delta of 1 Domains. Please contact AWS support to request an increase for this limit.
```